lib/pundit.rb

```diff
@@ -8,16 +8,18 @@
 require "active_support/core_ext/module/introspection"
 require "active_support/dependencies/autoload"
 
+# @api private
+# To avoid name clashes with common Error naming when mixing in Pundit,
+# keep it here with compact class style definition.
+class Pundit::Error < StandardError; end # rubocop:disable Style/ClassAndModuleChildren
+
 # @api public
 module Pundit
   SUFFIX = "Policy".freeze
 
   # @api private
   module Generators; end
 
-  # @api private
-  class Error < StandardError; end
-
   # Error that will be raised when authorization has failed
   class NotAuthorizedError < Error
```

Comment:
shouldn't this then be `< ::Pundit::Error` ?

Reply:
@bf4 Thanks for the comments - can you expand on this?

This does seem to inherit from the correct `Error` class if that's the concern:

```
pry(main)> require 'pundit'
=> true
pry(main)> Pundit::NotAuthorizedError.ancestors
=> [Pundit::NotAuthorizedError,
 Pundit::Error,
 StandardError,
 Exception,
 Object,
 JSON::Ext::Generator::GeneratorMethods::Object,
 PP::ObjectMixin,
 Kernel,
 BasicObject]
```